src/framebuffer.ml

```diff
@@ -102,6 +106,7 @@ let draw_line (x0 : int) (y0 : int) (x1 : int) (y1 : int) (col : int) (buffer :
   in loop x0 y0 initial_error
 
 let draw_polygon (points : (int * int) list) (col : int) (buffer : t) =
+  buffer.dirty <- true;
```

Comment:
This is too soon to decide in `draw_polygon`, as if the point list is empty we don't modify the buffer. In this instance this function doesn't modify the buffer directly, but calls draw_line, so I'd let that make the buffer as dirty.

Or put another way, only mark the buffer dirty when you know you differently change the contents: even when we call draw_line, draw_line might also decide not to do any work.